xrpl/ledger-entry-types/oracle.go

```diff
@@ -0,0 +1,78 @@
+package ledger
+
+import "github.com/Peersyst/xrpl-go/xrpl/transaction/types"
+
+// A PriceData object represents the price information for a token pair.
+type PriceData struct {
+	// The primary asset in a trading pair. Any valid identifier, such as a stock symbol,
+	// bond CUSIP, or currency code is allowed.
+	BaseAsset string
+	// The quote asset in a trading pair. The quote asset denotes the
+	//price of one unit of the base asset.
+	QuoteAsset string
+	// The asset price after applying the Scale precision level. It's not included if
+	// the last update transaction didn't include the BaseAsset/QuoteAsset pair.
+	AssetPrice uint64 `json:",omitempty"`
+	// The scaling factor to apply to an asset price. For example, if Scale is 6 and original price is 0.155,
+	//then the scaled price is 155000. Valid scale ranges are 0-10.
+	//It's not included if the last update transaction didn't include the BaseAsset/QuoteAsset pair.
+	Scale uint8 `json:",omitempty"`
+}
+
+// An Oracle ledger entry holds data associated with a single price oracle object.
+// Requires PriceOracle amendment.
+// Example:
+// ```json
+//
+//	{
+//	  "LedgerEntryType": "Oracle",
+//	  "Owner": "rNZ9m6AP9K7z3EVg6GhPMx36V4QmZKeWds",
+//	  "Provider": "70726F7669646572",
+//	  "AssetClass": "63757272656E6379",
+//	  "PriceDataSeries": [
+//	    {
+//	      "PriceData": {
+//	        "BaseAsset": "XRP",
+//	        "QuoteAsset": "USD",
+//	        "AssetPrice": 740,
+//	        "Scale": 3,
+//	      }
+//	    },
+//	  ],
+//	  "LastUpdateTime": 1724871860,
+//	  "PreviousTxnID": "C53ECF838647FA5A4C780377025FEC7999AB4182590510CA461444B207AB74A9",
+//	  "PreviousTxnLgrSeq": 3675418
+//	}
+//
+// ```
+type Oracle struct {
```

Comment:
Do we also need the serialization/deserialization tests like what we have for other structs ?
For all the structs you added in this PR.

Reply:
These methods are not used by any library component atm. If necessary, we will add them later